### ISSUE TITLE
Clarify "appropriate annotations preserved" when slicing a SeqRecord

### ISSUE THREAD

**Issue by althonos** [biopython/biopython#1738]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
>>> import sys; print(sys.version)
3.6.6 (default, Jun 27 2018, 13:11:40) 
[GCC 8.1.1 20180531]
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
Linux-4.17.6-1-ARCH-x86_64-with-arch
>>> import Bio; print(Bio.__version__)
1.72
```

### Expected behaviour

When slicing a `SeqRecord` instance, keep annotations of type `source` even when they overflow the slicing window. For instance, given a `SeqRecord` annotated coming completely from *E. coli*, slicing the record should return another `SeqRecord` also annotated with a *E. coli* source.

### Actual behaviour

Source annotation is lost when not completely contained in the slicing window.

### Steps to reproduce

```python
from Bio.SeqRecord import SeqRecord
from Bio.Seq import Seq
from Bio.SeqFeature import SeqFeature, FeatureLocation

seq = Seq('AAAAATGCAAAA')
loc = FeatureLocation(0, len(seq)) 
ftr = SeqFeature(loc, 'source', qualifiers={'organism': ['E. coli']})

rec = SeqRecord(seq, id="test", features=[ftr])

print(rec.features)
print(rec[:].features)
print(rec[4:8].features)
```

This will print the source feature twice and an empty list once, since the `source` annotation is not carried over when slicing the record.

**Comment by althonos** [comment]
Ok, fair enough. I just felt that the "appropriate annotations preserved" also applied to the `source` feature as well, but I can get why you don't want any feature-specific behaviour.

**Comment by peterjc** [comment]
Good point - my meaning with "appropriate annotations preserved" isn't necessarily going to be what the reader expects. We can try to improve that, reopening for a documentation clarification.

**Comment by SchwarzMarek** [comment]
Hello,
I've needed to do some visualization on sliced sequences and I wanted to have features spanning bounds of the slice preserved. Since this behaviour is not supported by standard slice I've wrote function for this kind of slicing.

My feeling is, that I'm not the only one with need for such slicing function (not to replace the standard one). I'm considering opening a PR, but the code need some additional work (cleaning, etc), which I'm willing to do, if the PR would be welcome.

Best regards
Marek
